Separate system tests out of E2E

mkolesnik opened the pull request:
This PR separates the system tests out of E2E, making E2E super simple and system tests more focused.

As part of this work the E2E tests have also been improved somewhat.

Please see individual commits for more details.

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

mkolesnik committed:
Move test scripts to "test" directory

In order to split out system test scripts, it makes sense to rename the
directory.

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

mkolesnik committed:
Remove white box system testing from E2E

This will be readded cleanly in the following commit(s) but for now
simplify E2E testing by removing the code.

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

mkolesnik committed:
Change execution order in E2E

Run the specific tests (using Shipyard's E2E script) first.

This makes sure we only deploy the environment once, and don't try to
redeploy unnecessarily.

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

mkolesnik committed:
Add system tests (scripts) as proper system tests

The system tests are now in a proper script, and can be called with
`make system`.
The tests will be covered by GHA which covers the relevant tested
matrix (currently only no/globalnet).

Being white box specific tests, they utilize their own settings which
deploy a single node cluster for faster and lighter deploys.

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

dfarrell07 commented on .github/workflows/system.yml:
```diff
@@ -0,0 +1,28 @@
+---
+name: System tests
```
This ends up showing in the UI of all PRs, and currently it's Leading Caps in all workflows in all repos I think. Would be good to be consistent.

```suggestion
name: System Tests
```

Also, I wonder if anyone has a suggestion for a more descriptive workflow/job name than "System Tests/Test"? I get the idea, but it took some thinking and I authored a lot of these tests. :stuck_out_tongue_winking_eye: A more verbose job name maybe? /me is struggling to think of a good suggestion, will follow up if I think of one

mkolesnik replied:
Updated the caps, I'm not married to the name if you have a better suggestion I don't mind changing it

tpantelis replied:
Unfortunately only the job name is displayed in the branch settings tab so having that more descriptive helps. I suggest job name "_Deployment Test_", so full  name "_System Tests/Deployment Test_".

tpantelis commented on .github/workflows/system.yml:
```diff
@@ -0,0 +1,28 @@
+
+jobs:
+  system:
+    name: Test
```
Unfortunately only the job name is displayed in the branch settings tab so having that more descriptive helps. I suggest job name "_Deployment Test_", so full name "_System Tests/Deployment Test_".

```suggestion
    name: Deployment Test
```

mkolesnik committed:
Use smaller KIND clusters

As we're basically running standard E2E without any special
requirements, we can use smaller clusters and save some time and
resources when deploying with KIND.

Signed-off-by: Mike Kolesnik <mkolesni@redhat.com>

dfarrell07 commented:
> Error: two kubecontexts must be specified

mkolesnik commented:
> > Error: two kubecontexts must be specified

Right, I was foolishly trying to avoid parsing the settings file to get the clusters :)

I amended the necessary commit to add code to parse the settings properly, and while at it squashed another commit (the rename) into the one above it.

The PR is otherwise unchanged.
